pkg/network/http/http_stats.go

```diff
@@ -26,53 +45,87 @@ type RequestStats [5]struct {
 	// Note: every time we add a latency value to the DDSketch below, it's possible for the sketch to discard that value
 	// (ie if it is outside the range that is tracked by the sketch). For that reason, in order to keep an accurate count
 	// the number of http transactions processed, we have our own count field (rather than relying on DDSketch.GetCount())
-	count     int
-	latencies *ddsketch.DDSketch
+	Count     int
+	Latencies *ddsketch.DDSketch
+
+	// This field holds the value (in milliseconds) of the first HTTP request
+	// in this bucket. We do this as optimization to avoid creating sketches with
+	// a single value. This is quite common in the context of HTTP requests without
+	// keep-alives where a short-lived TCP connection is used for a single request.
+	FirstLatencySample float64
```

Comment:
Is it worth implementing a hacky union type where we only store one of `Latencies` or `FirstLatencySample` in an `interface{}` and use `Count` to assume the type ?

Reply:
humm, I honestly don't see a lot of benefit in doing that. I think if we are trying to save 8 bytes using an interface would defeat the purpose of this change since the interface value itself has 2 words already.